api/odl.py

```diff
@@ -539,29 +540,24 @@ def fulfill(self, patron, pin, licensepool, internal_format, **kwargs):
         loan = loan.one()
         return self._fulfill(loan, internal_format)
 
-    def _fulfill(
-        self,
-        loan: Loan,
-        delivery_mechanism: Optional[LicensePoolDeliveryMechanism] = None,
-    ) -> FulfillmentInfo:
-        licensepool = loan.license_pool
-        doc = self.get_license_status_document(loan)
-        status = doc.get("status")
-
-        if status not in [self.READY_STATUS, self.ACTIVE_STATUS]:
-            # This loan isn't available for some reason. It's possible
-            # the distributor revoked it or the patron already returned it
-            # through the DRM system, and we didn't get a notification
-            # from the distributor yet.
-            self.update_loan(loan, doc)
-            raise CannotFulfill()
-
-        expires = doc.get("potential_rights", {}).get("end")
-        expires = dateutil.parser.parse(expires)
-
-        links = doc.get("links", [])
+    @staticmethod
+    def _find_content_link_and_type(
+        links: List[Dict],
+        delivery_mechanism: Union[
+            Optional[str], Optional[LicensePoolDeliveryMechanism]
```

Comment:
Do we ever get a `str` passed into this function now? It looks like its only ever called with `LicensePoolDeliveryMechanism`.

Reply:
Seems like the type hint on `_fulfill` is wrong then, it looks like `_find_content_link_and_type` is only called from there, and its type hint is `Optional[LicensePoolDeliveryMechanism]`. Thats what made me think we weren't expecting a string to get to `_find_content_link_and_type`.

I guess this type hint was wrong before, so we can leave it be.